x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeProcessor.java

```diff
@@ -75,32 +70,15 @@ DateTimeExtractor extractor() {
     }
 
     @Override
-    public Object process(Object l) {
-        if (l == null) {
-            return null;
-        }
-
-        ReadableDateTime dt;
-        if (l instanceof String) {
-            // 6.4+
-            final long millis = Long.parseLong(l.toString());
-            dt = new DateTime(millis, DateTimeZone.forTimeZone(timeZone));
-        } else if (l instanceof ReadableInstant) {
-            // 6.3-
-            dt = (ReadableDateTime) l;
-            if (!TimeZone.getTimeZone("UTC").equals(timeZone)) {
-                dt = dt.toDateTime().withZone(DateTimeZone.forTimeZone(timeZone));
-            }
-        } else {
-            throw new SqlIllegalArgumentException("A string or a date is required; received {}", l);
-        }
+    public Object doProcess(long millis) {
+        ReadableDateTime dt = new DateTime(millis, DateTimeZone.forTimeZone(timeZone()));
```

Comment:
With the introduction of the abstract class, in case of `6.3-` we already have a `ReadableDateTime` which we now convert to `millis` and then back to `ReadableDateTime`. Could this be a perf issue?

Reply:
Since this is operating on the data returned by the search it isn't as much of a problem as it could be. And it'll only come up during the rolling upgrade too. Since we're using the opportunity to switch from Joda time classes to Java 8 time classes I'm on board with it.